app/services/bgs/dependent_service.rb

```diff
@@ -81,32 +81,38 @@ def submit_686c_form(claim)
         # why I am deliberately raising these errors here.
         validate_file_number_format!(file_number:)
         validate_file_number_matches_ssn!(file_number:)
-        if claim.submittable_686?
-          if Flipper.enabled?(:dependents_encrypt_jobs)
-            BGS::SubmitForm686cEncryptedJob.perform_async(
-              uuid, @icn, claim.id, encrypted_vet_info
-            )
-          else
-            BGS::SubmitForm686cJob.perform_async(
-              uuid, @icn, claim.id, form_hash_686c
-            )
-          end
-        else
-          if Flipper.enabled?(:dependents_encrypt_jobs) # rubocop:disable Style/IfInsideElse
-            BGS::SubmitForm674EncryptedJob.perform_async(
-              uuid, @icn, claim.id, encrypted_vet_info
-            )
-          else
-            BGS::SubmitForm674Job.perform_async(
-              uuid, @icn, claim.id, form_hash_686c
-            )
-          end
-        end
+        submit_form_job_id = if claim.submittable_686?
+                               if Flipper.enabled?(:dependents_encrypt_jobs)
+                                 BGS::SubmitForm686cEncryptedJob.perform_async(
+                                   uuid, @icn, claim.id, encrypted_vet_info
+                                 )
+                               else
+                                 BGS::SubmitForm686cJob.perform_async(
+                                   uuid, @icn, claim.id, form_hash_686c
+                                 )
+                               end
+                             else
+                               if Flipper.enabled?(:dependents_encrypt_jobs) # rubocop:disable Style/IfInsideElse
+                                 BGS::SubmitForm674EncryptedJob.perform_async(
+                                   uuid, @icn, claim.id, encrypted_vet_info
+                                 )
+                               else
+                                 BGS::SubmitForm674Job.perform_async(
+                                   uuid, @icn, claim.id, form_hash_686c
+                                 )
+                               end
+                             end
         Rails.logger.info('BGS::DependentService succeeded!', { user_uuid: uuid, saved_claim_id: claim.id, icn: })
       end
+
+      {
+        submit_pdf_job_id:,
+        submit_form_job_id:
+      }
     rescue => e
       Rails.logger.error('BGS::DependentService failed!', { user_uuid: uuid, saved_claim_id: claim.id, icn:, error: e.message }) # rubocop:disable Layout/LineLength
       log_exception_to_sentry(e, { icn:, uuid: }, { team: Constants::SENTRY_REPORTING_TEAM })
+      raise e
```

Comment:
Found that an error in the async jobs being performed would be logged but not raised to the controller calling it.

Reply:
I wouldn't expect errors in an async job to raise errors in the controller. What type of error would be caught here? I would only expect an error if the number of params is wrong or something like that. The spec setup is so complex and this method so long that it's unclear to me whether the errors are coming from the async jobs or from something else. It would surprise me if the error is coming from a BGS gem method like `find_person_by_ptcpnt_id` instead of the async jobs.

Reply:
Yeah, I misspoke there. Not the async jobs themselves since those would be happening in a different thread but the large amount of validation and fetching happening earlier in this method can raise errors. There's a lot of guarding of edge cases happening that will stop everything such as the file number not being equal to the ssn in `validate_file_number_format!`and `validate_file_number_matches_ssn!` and those errors should be raised up to the controller so the user we can tell the user that it didn't get submitted